src/h5web/visualizations/line/DataLine.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useMemo, useCallback } from 'react';
+import { useThree } from 'react-three-fiber';
+import { Vector3 } from 'three';
+import { scaleLinear } from 'd3-scale';
+import shallow from 'zustand/shallow';
+import { useLineVisConfig } from './config';
+import { Domain } from '../shared/models';
+
+interface Props {
+  data: number[];
+}
+
+function DataLine(props: Props): JSX.Element {
+  const { data } = props;
+  const {
+    size: { width, height },
+  } = useThree();
+
+  const [abscissaDomain, ordinateDomain] = useLineVisConfig(
+    state => [state.abscissaDomain, state.ordinateDomain],
+    shallow
+  );
+
+  const xScale = scaleLinear()
+    .domain(abscissaDomain as Domain)
+    .range([-width / 2, width / 2]);
+  const yScale = scaleLinear()
+    .domain(ordinateDomain as Domain)
+    .range([-height / 2, height / 2]);
+
+  const points = useMemo(() => {
+    return data.map((val, index) => new Vector3(xScale(index), yScale(val), 0));
+  }, [data, xScale, yScale]);
+
+  return (
+    <line>
+      <bufferGeometry
+        attach="geometry"
+        onUpdate={useCallback(self => self.setFromPoints(points), [points])}
```

Comment:
I cannot put `points` as a props so I have to use this syntax to call the `setFromPoints` 
 method from Three.js of `bufferGeometry`